src/Fluid/Flu_Close.cpp

```diff
@@ -377,26 +381,27 @@ bool Unphysical( const real Fluid[], const real Gamma_m1, const int CheckMinEngy
          Fluid[DENS] < (real)MIN_DENS  )
       return true;
 
-   if ( CheckMinEngyOrPres == CheckMinEngy  &&  Fluid[ENGY] < (real)MIN_PRES )
+   if ( CheckMode == CheckMinEtot  &&  Fluid[ENGY] < (real)MIN_EINT )
       return true;
 
-   if ( CheckMinEngyOrPres == CheckMinPres  &&
+   if ( CheckMode == CheckMinEint  &&
          (
-//          when the dual-energy formalism is adopted, do NOT calculate pressure from "Etot-Ekin" since it would suffer
+//          when adopting the dual-energy formalism, do NOT calculate pressure from "Etot-Ekin" since it would suffer
 //          from large round-off errors
-//          currently we use TINY_NUMBER as the floor value of entropy and hence here we use 2.0*TINY_NUMBER to validate entropy
-//          --> in general, for MIN_PRES > 0.0, we expect that unphysical entropy would lead to unphysical pressure
-//          --> however, the check "Fluid[ENPY] < (real)2.0*TINY_NUMBER" is necessary when MIN_PRES == 0.0
+//          --> currently we use TINY_NUMBER as entropy floor and hence here we use 2.0*TINY_NUMBER to
+//              validate entropy
+//          --> in general, MIN_PRES > 0.0 should be sufficient for detecting unphysical entropy
+//          --> however, the additional check "Fluid[ENPY] < (real)2.0*TINY_NUMBER" is necessary when MIN_PRES == 0.0
 #           if   ( DUAL_ENERGY == DE_ENPY )
-            Hydro_DensEntropy2Pres( Fluid[DENS], Fluid[ENPY], Gamma_m1, CorrPres_No, NULL_REAL ) < (real)MIN_PRES  ||
+            Hydro_DensEntropy2Pres( Fluid[DENS], Fluid[ENPY], EoS_AuxArray[1], NoFloor, NULL_REAL ) < (real)MIN_PRES  ||
```

Comment:
Do we here still check the pressure, rather than Eint?

Reply:
It still checks pressure since the `DE_ENPY` mode only works for the constant-gamma EoS anyway. I plan to implement the `DE_EINT` mode (i.e., solving the internal energy equation instead of entropy equation) to support general EoS, if necessary.